cas-template.md

```diff
@@ -0,0 +1,39 @@
+# OCI CAS Template Protocol
+
+This is version 1 of this specification.
+
+The CAS-template protocol is configured via a single [URI Template][rfc6570].
+When configured via a [`casEngines` entry](ref-engine-discovery.md#ref-engines-objects), the `uri` property MUST be set, and its value is the URI Template.
+
+For a given blob digest, consumers MUST provide at least the following variables:
+
+* `digest`, matching `digest` in the [`digest` rule][digest].
+* `algorithm`, matching `algorithm` in the `digest` rule.
+* `encoded`, matching `encoded` in the `digest` rule.
+
```

Comment:
I'm not sure if these variables SHOULD be restricted in spec. E.g. a CAS URI:
```
https://example.com/oci-cas/ubuntu/14.04
```
Which returns pure CAS object, but the URI path doesn't conform to what you defined. You want to reject it? 
And I really se no reason to restrict it. This URI is provided by image provider, they can use any template they like. 
You restrict these, if you want to parse them from URI? But how to parse?

Reply:
I feel we can allow consumers to define URI with unlimited template. If I miss something?

Reply:
> I'm not sure if these variables SHOULD be restricted in spec. E.g. a CAS URI:
> ```
> https://example.com/oci-cas/ubuntu/14.04
> ```
> Which returns pure CAS object (OCI index media type), but the URI path doesn't conform to what you defined. You want to reject it? 
> And I really se no reason to restrict it. This URI is provided by image provider, they can use any template they like. 
> You restrict these, if you want to parse them from URI? But how to parse?
> I feel we can allow consumers to define URI with unlimited template. If I miss something?

Oh, I really missed something. You define `oci-cas-template-v1` for REF and CAS protocol, and allow consumers to extend the favorite protocols. Ignore my above questions, they make no sense.

Reply:
> You define `oci-cas-template-v1` for REF and CAS protocol...

nit: `oci-cas-template-v1` is just for CAS.  `oci-index-template-v1` is for refs.